### ISSUE TITLE
Build Error: execvp: protoc-c: Not a directory

### ISSUE THREAD

**Issue by wenhuizhang** [checkpoint-restore/criu#2060]
Build Error for HEAD detached at v3.17.1

```
root@n223-247-006:~/criu# make
Note: Building without setproctitle() and strlcpy() support.
      To enable these features, please install libbsd-devel (RPM) / libbsd-dev (DEB).
Note: Building criu without amdgpu_plugin.
Note: libdrm and libdrm_amdgpu are required to build amdgpu_plugin.
Note: Building without GnuTLS support
Makefile.config:53: Warn: you have no libnftables installed
Makefile.config:54: Warn: Building without nftables support
grep: images/google/protobuf/descriptor.proto: No such file or directory
  PBCC     images/rseq.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/apparmor.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/bpfmap-data.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fown.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/img-streamer.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/timens.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/macvlan.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/autofs.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sysctl.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/time.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/binfmt-misc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/seccomp.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pidns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/userns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/cgroup.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/rpc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/siginfo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/rlimit.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/file-lock.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/creds.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/utsns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ipc-desc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ipc-var.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sk-opts.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/timer.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pipe-data.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sk-packet.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pstree.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fs.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/remap-file-path.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fifo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/cpuinfo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/stats.pb-c.c
make[1]: execvp: protoc-c: Not a directory
make[1]: Nothing to be done for 'all'.
grep: images/google/protobuf/descriptor.proto: No such file or directory
  PBCC     images/rseq.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/apparmor.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/bpfmap-data.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fown.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/img-streamer.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/timens.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/macvlan.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/autofs.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sysctl.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/time.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/binfmt-misc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/seccomp.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pidns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/userns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/cgroup.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/rpc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/siginfo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/rlimit.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/file-lock.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/creds.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/utsns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ipc-desc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ipc-var.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sk-opts.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/timer.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pipe-data.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sk-packet.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pstree.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fs.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/remap-file-path.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fifo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/cpuinfo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/stats.pb-c.c
make[1]: execvp: protoc-c: Not a directory
make[1]: *** No rule to make target 'images/stats.pb-c.c', needed by 'images/stats.o'.  Stop.
make: *** [Makefile:217: images/built-in.o] Error 2
```

**Comment by adrianreber** [comment]
You need to install CRIU's prerequisites. This error comes from the missing protobuf-c compiler.

**Comment by wenhuizhang** [comment]
> You need to install CRIU's prerequisites. This error comes from the missing protobuf-c compiler.

Hi, Sorry for bothering , however I installed it , and still got the same error. 
`sudo apt install protobuf-compiler python3-protobuf libc6 libgnutls30 libnftables1`

Error log:

```
root@n223-247-006:~/criu# make
Note: Building criu without amdgpu_plugin.
Note: libdrm and libdrm_amdgpu are required to build amdgpu_plugin.
Note: Building without GnuTLS support
Makefile.config:53: Warn: you have no libnftables installed
Makefile.config:54: Warn: Building without nftables support
  PBCC     images/rseq.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/apparmor.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/bpfmap-data.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fown.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/img-streamer.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/timens.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/google/protobuf/descriptor.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/macvlan.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/autofs.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sysctl.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/time.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/binfmt-misc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/seccomp.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pidns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/userns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/cgroup.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/rpc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/siginfo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/rlimit.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/file-lock.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/creds.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/utsns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ipc-desc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ipc-var.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sk-opts.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/timer.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pipe-data.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sk-packet.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pstree.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fs.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/remap-file-path.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fifo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/cpuinfo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/stats.pb-c.c
make[1]: execvp: protoc-c: Not a directory
make[1]: Nothing to be done for 'all'.
  PBCC     images/rseq.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/apparmor.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/bpfmap-data.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fown.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/img-streamer.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/timens.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/google/protobuf/descriptor.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/macvlan.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/autofs.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sysctl.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/time.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/binfmt-misc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/seccomp.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pidns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/userns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/cgroup.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/rpc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/siginfo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/rlimit.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/file-lock.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/creds.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/utsns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ipc-desc.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ipc-var.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sk-opts.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/timer.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pipe-data.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/sk-packet.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/pstree.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fs.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/remap-file-path.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/fifo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/ns.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/cpuinfo.pb-c.c
make[1]: execvp: protoc-c: Not a directory
  PBCC     images/stats.pb-c.c
make[1]: execvp: protoc-c: Not a directory
make[1]: *** No rule to make target 'images/stats.pb-c.c', needed by 'images/stats.o'.  Stop.
make: *** [Makefile:217: images/built-in.o] Error 2
```

**Comment by adrianreber** [comment]
No, you installed the protobuf-compiler not the protobuf-c-compiler.

There is a specia protobufl package to support C.

**Comment by rst0git** [comment]
@wenhuizhang A list of all build dependencies is available at https://criu.org/Installation.

**Comment by wenhuizhang** [comment]
> @wenhuizhang A list of all build dependencies is available at https://criu.org/Installation.

Thanks so much for your swift reply. 

I installed all packages listed in the Installation website, it seems like I still miss something. 

I followed the steps below:

```
apt --fix-broken install
apt install protobuf-compiler libc6 libgnutls30 libnftables1  
apt-get install  protobuf-c-compiler
apt-get install libprotobuf-dev libprotobuf-c-dev protobuf-c-compiler protobuf-compiler python3-protobuf
apt-get install pkg-config python-ipaddress libbsd-dev iproute2  libnftables-dev libcap-dev   libnl-3-dev libnet-dev
apt-get install libaio-dev libgnutls28-dev python3-future 
apt-get install asciidoc


make
make install
```

The `make` gives me no error, however, when I do `make install`, I got error messages as below, I am checking if there are some python packages I missed .

```
root@n223-247-006:~/criu# make install
Note: Building criu without amdgpu_plugin.
Note: libdrm and libdrm_amdgpu are required to build amdgpu_plugin.
  GEN      footer.txt
  GEN      crit.1
Note: Writing crit.1
  GEN      criu-ns.1
Note: Writing criu-ns.1
  GEN      compel.1
Note: Writing compel.1
  GEN      amdgpu_plugin.1
Note: Writing amdgpu_plugin.1
  GEN      criu.8
Note: Writing criu.8
  INSTALL  criu.8
  INSTALL  crit.1 criu-ns.1 compel.1 amdgpu_plugin.1
make[1]: Nothing to be done for 'all'.
make[1]: 'images/built-in.o' is up to date.
make[1]: 'compel/plugins/std.lib.a' is up to date.
make[1]: 'compel/plugins/fds.lib.a' is up to date.
make[1]: 'compel/libcompel.a' is up to date.
make[1]: 'compel/compel-host-bin' is up to date.
make[1]: Nothing to be done for 'all'.
make[1]: 'soccr/libsoccr.a' is up to date.
make[2]: Nothing to be done for 'all'.
make[2]: Nothing to be done for 'all'.
make[2]: Nothing to be done for 'all'.
make[2]: Nothing to be done for 'all'.
make[1]: Nothing to be done for 'all'.
make[2]: 'lib/c/built-in.o' is up to date.
make[2]: 'lib/c/built-in.o' is up to date.
make[2]: Nothing to be done for 'lib/c/criu.pc.in'.
  INSTALL  lib
  INSTALL  pkgconfig/criu.pc
  INSTALL  crit
Traceback (most recent call last):
  File "scripts/crit-setup.py", line 2, in <module>
    from distutils.core import setup
ModuleNotFoundError: No module named 'distutils.core'
make[1]: *** [lib/Makefile:62: install] Error 1
make: *** [Makefile.install:37: install-lib] Error 2
```

**Comment by wenhuizhang** [comment]
Hi, I found that it is due to disutils deprecated in python3  newer versions, after downgrade the version , it worked through. Thanks so much for your help!

**Comment by rst0git** [comment]
>   File "scripts/crit-setup.py", line 2, in <module>
    from distutils.core import setup
ModuleNotFoundError: No module named 'distutils.core'

Thank you for pointing this out! We have a fix for this problem in this pull request: https://github.com/checkpoint-restore/criu/pull/2009.

**Comment by wenhuizhang** [comment]
> 



> > File "scripts/crit-setup.py", line 2, in 
> > from distutils.core import setup
> > ModuleNotFoundError: No module named 'distutils.core'
> 
> Thank you for pointing this out! We have a patch that fixes this problem in: #2009.

Thanks so much! hope this patch could be merge and applied to other previous versions as well:) 
I am closing this issue now . :)
